Add iOS version of Dolphin Old

WilliamLCobb opened the pull request:
Most of the changes are pretty straightforward. 

Added iOS directory in Source.
Added new IOS definition when building for iOS
Added build steps for in readme.

Cmake will generate an Xcode project in the build directory and copy in the Xcode project and workspace from the iOS directory. The workspace will combine these two projects, allowing full compatibility with future cmake and source changes while also giving control of all of the build settings for iOS. Since iOS is not inherently supported by cmake, this would have been very hard if not impossible without this step. The iOS app can be built in the Xcode GUI and also built from the command line. 

Only one commit has a chance of affecting other builds: "IOS macro workaround" (0a22fb7). IOS was being used as a variable in a struct in Network.cpp and Network.h. Since I was using this as a definition for building iOS, i changed all instances of this variable to _IOS. This probably needs to be changed to something else, I just wasn't sure what

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dolphin-emu/dolphin/3881)
<!-- Reviewable:end -->

Helios747 commented:
@dolphin-emu-bot rebuild

phire commented on Source/iOS/DolphiniOS.xcodeproj/project.xcworkspace/contents (Will’s MacBook Pro's conflicted copy 2016-06-03).xcworkspacedata:
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
This file shouldn't be here.

phire commented on Source/Core/Common/Network.h:
```diff
@@ -11,7 +11,7 @@
 {
 	BBA = 0,
-	IOS = 1
+	_IOS = 1
```
can we rename it to something better, instead of just tacking on an underscore?

JosJuice commented:
Could you call the define something other than `IOS`? It's easy to think it refers to IOS (the software running on the Wii's ARM CPU) rather than iOS, and it also causes that problem with the `IOS` enum value that you mentioned. Would `APPLE_IOS` work?

The extraction/decompression code needs to be removed completely from the PR, not committed and then reverted.

WilliamLCobb commented:
Yea APPLE_IOS is fine

phire commented on Source/Core/Common/GL/GLInterface/GLInterface.cpp:
```diff
@@ -6,7 +6,9 @@
 
-#if defined(__APPLE__)
+#if defined IOS
+#include "Common/GL/GLInterface/IGL.h"
```
As discussed on IRC. You should stick with the official apple name of [EAGL](https://developer.apple.com/library/ios/documentation/OpenGLES/Reference/EAGL_functions/) instead of making up a new one.

WilliamLCobb replied:
Will do

WilliamLCobb commented:
Close this please, I'll create another pr without decompressors in commit history

degasus commented:
Reviewed 24 of 301 files at r1.
Review status: 23 of 246 files reviewed at latest revision, 20 unresolved discussions, some commit checks failed.

---

*[Readme.md, line 110 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWs_ZPwAdD2yJm_dHS:-KJWs_ZPwAdD2yJm_dHT:211576953) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Readme.md#L110)):*
> ```Markdown
> 1. `mkdir Build`
> 2. `cd Build`
> 3. `cmake .. -G Xcode -DENABLE_PCH=false -DCMAKE_TOOLCHAIN_FILE=../Source/iOS/ios.toolchain.cmake`
> ```

-DENABLE_PCH should be enforced by the cmake file is required.

---

*[Externals/SOIL/SOIL.c, line 25 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWskVZpWNg8mYV2YFO:-KJWskVZpWNg8mYV2YFP:1399443268) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Externals/SOIL/SOIL.c#L25)):*
> ```C
> 		#include <GL/gl.h>
> 	#endif
> #elif ((defined(__APPLE__) || defined(__APPLE_CC__)) && !IOS)
> ```

What do you think about defininf __APPLE__ on IOS?

---

*[Source/Core/Common/Arm64Emitter.cpp, line 15 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWt3v_t0MZqU_SKKG2:-KJWt3v_t0MZqU_SKKG3:1380768159) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/Arm64Emitter.cpp#L15)):*
> ```cpp
> 
> #if IOS
>     #include <array>
> ```

no need to hide array within IOS. It's either needed for all, or non.

---

*[Source/Core/Common/ArmCPUDetect.cpp, line 66 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWtSxGV1xk1DZGa0CX:-KJWtSxGV1xk1DZGa0CY:1773417052) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/ArmCPUDetect.cpp#L66)):*
> ```cpp
> 	strncpy(cpu_string, GetCPUString().c_str(), sizeof(cpu_string));
> #if !IOS
> 	unsigned long hwcaps = getauxval(AT_HWCAP);
> ```

I think we should try to support this. 

---

*[Source/Core/Common/CDUtils.cpp, line 73 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWtgXRHhM-m2cVkyai:-KJWtgXRHhM-m2cVkyaj:970408000) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/CDUtils.cpp#L73)):*
> ```cpp
> {
>     std::vector<std::string> drives;
>     return drives;
> ```

tab vs space
return {};

---

*[Source/Core/Common/CMakeLists.txt, line 74 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWtwdXmrWsXhBYIx-I:-KJWtwdXmrWsXhBYIx-J:985344557) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/CMakeLists.txt#L74)):*
> ```CMake
> 	GL/GLInterface/GLInterface.cpp)
> 
> if(USE_EGL AND NOT IOS)
> ```

Just configure USE_EGL=False within cmake

---

*[Source/Core/Common/MemArena.cpp, line 105 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWuGTNH_gJBXjxb4Rf:-KJWuGTNH_gJBXjxb4Rg:-1990138430) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/MemArena.cpp#L105)):*
> ```cpp
>     int flags;
> #if IOS
>     flags = MAP_ANON | MAP_PRIVATE | ((base == nullptr) ? 0 : MAP_FIXED);
> ```

maybe a comment why you've choosen this flags

---

*[Source/Core/Common/Network.h, line 11 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWuijFfdyTdLU6KNge:-KJWuijFfdyTdLU6KNgf:-833600659) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/Network.h#L11)):*
> ```objc
> #include "Common/CommonTypes.h"
> 
> enum MACConsumer
> ```

Will an enum class help here?

---

*[Source/Core/Common/GL/GLInterface/GLInterface.cpp, line 9 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWvBJgdjY2iUQTJioz:-KJWvBJhYv2ZqMxLb7ap:2073871136) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/GL/GLInterface/GLInterface.cpp#L9)):*
> ```cpp
> #include "Common/GL/GLInterfaceBase.h"
> 
> #if defined IOS
> ```

defined(IOS)

---

*[Source/Core/Common/GL/GLInterface/GLInterface.cpp, line 31 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWvEjMIqoZA5NyY-cd:-KJWvEjMIqoZA5NyY-ce:-1775187424) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/GL/GLInterface/GLInterface.cpp#L31)):*
> ```cpp
> std::unique_ptr<cInterfaceBase> HostGL_CreateGLInterface()
> {
>     #if defined IOS
> ```

spaces vs tab

---

*[Source/Core/Common/GL/GLInterface/IGL.h, line 26 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWvdRvq8ATHxXjINkK:-KJWvdRvq8ATHxXjINkL:937975804) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/GL/GLInterface/IGL.h#L26)):*
> ```objc
> 	virtual void ShutdownPlatform() {}
> 
> public:
> ```

please add public first, then protected, then private

---

*[Source/Core/Common/GL/GLInterface/IGL.mm, line 19 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWwF2pBgqYD60y747n:-KJWwF2pBgqYD60y747o:424582323) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/GL/GLInterface/IGL.mm#L19)):*
> ```XML
> 
> 
> EAGLContext *context;
> ```

member variables instead of static please.

---

*[Source/Core/Common/GL/GLInterface/IGL.mm, line 53 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWwepcwVSkF22COGXV:-KJWwepcwVSkF22COGXW:-65833916) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/GL/GLInterface/IGL.mm#L53)):*
> ```XML
>     //eaglLayer.drawableProperties = @{kEAGLDrawablePropertyRetainedBacking : @(YES)};
>     
>     s_backbuffer_width = glkView.frame.size.width * 2;
> ```

why times 2?

---

*[Source/Core/Common/Logging/ConsoleListeneriOS.cpp, line 16 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWxRVRttXYeIr4TfM6:-KJWxRVRttXYeIr4TfM7:-830408417) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/Logging/ConsoleListeneriOS.cpp#L16)):*
> ```cpp
> }
> 
> void ConsoleListener::Log(LogTypes::LOG_LEVELS level, const char* text)
> ```

I wouldn't call this "ios", more like a generic one. If there isn't one (I haven't checked), please just rename this one to "Generic"

---

*[Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp, line 20 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWzyspW5RDyVAmdR9c:-KJWzyspW5RDyVAmdR9d:-229638398) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp#L20)):*
> ```cpp
> using namespace Arm64Gen;
> 
> #if !IOS
> ```

Did this functions fail to compile, or just fail at execution? I'd like to see fastmem at some point :/

---

*[Source/Core/InputCommon/CMakeLists.txt, line 16 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX-OYsLp0_KLy6ksFL:-KJX-OYsLp0_KLy6ksFM:-1579019434) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/InputCommon/CMakeLists.txt#L16)):*
> ```CMake
> 				ControllerInterface/ForceFeedback/ForceFeedbackDevice.cpp)
> elseif(IOS)
> 	add_definitions(-DCIFACE_USE_IOS)
> ```

Please move this to ControllerInterface.h

---

*[Source/Core/InputCommon/CMakeLists.txt, line 18 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX-CZitHrBiFexbgOL:-KJX-CZj3vkstPo_6cyA:-1775187424) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/InputCommon/CMakeLists.txt#L18)):*
> ```CMake
> 	add_definitions(-DCIFACE_USE_IOS)
> 	set(SRCS ${SRCS}
> 	         	ControllerInterface/iOS/iOS.mm
> ```

spaces vs tab

---

*[Source/Core/VideoBackends/OGL/PostProcessing.cpp, line 52 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX-UtrqktQHO8gRfJY:-KJX-UtrqktQHO8gRfJZ:-197028094) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/VideoBackends/OGL/PostProcessing.cpp#L52)):*
> ```cpp
>     GLint defaultFBO;
>     glGetIntegerv(0x8CA6, &defaultFBO);
>     glBindFramebuffer(GL_DRAW_FRAMEBUFFER, defaultFBO);
> ```

GLInterface->GetDefaultFramebuffer() ?

---

*[Source/Core/VideoBackends/OGL/Render.cpp, line 1340 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX-chYEZCef5qMfbuJ:-KJX-chYEZCef5qMfbuK:1639331110) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/VideoBackends/OGL/Render.cpp#L1340)):*
> ```cpp
> 			// Tell the OSD Menu about the current internal resolution
> 			OSDInternalW = xfbSource->sourceRc.GetWidth(); OSDInternalH = xfbSource->sourceRc.GetHeight();
>             GLInterface->Prepare();
> ```

Maybe within BlitScreen?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881)*
<!-- Sent from Reviewable.io -->

JosJuice commented:
@WilliamLCobb It is possible to close the PR on your own. Or, you could force push the new version to this PR's branch, avoiding the need for a second PR.

lioncash commented:
Review status: 23 of 246 files reviewed at latest revision, 46 unresolved discussions, some commit checks failed.

---

*[Source/Core/Common/GL/GLInterface/IGL.mm, line 25 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX8_DIsE247xt8BRtp:-KJX8_DIsE247xt8BRtq:1368472124) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/Common/GL/GLInterface/IGL.mm#L25)):*
> ```XML
> void cInterfaceIGL::Swap()
> {
>     //NSLog(@"Frame: %d", framenum++);
> ```

Please don't leave commented code all over the place.

---

*[Source/Core/Common/GL/GLInterface/IGL.mm, line 30 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX8e3i4KW5q5Xm71Oo:-KJX8e3jZP_xo1YQv7KH:424678959) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/Common/GL/GLInterface/IGL.mm#L30)):*
> ```XML
> void cInterfaceIGL::SwapInterval(int Interval)
> {
>     
> ```

Unnecessary whitespace (and in other places)

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.h, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX93uuIda82-nsDq12:-KJX93uuIda82-nsDq13:676036456) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.h#L1)):*
> ```objc
> // Copyright 2008 Dolphin Emulator Project
> ```

2016

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.h, line 12 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX96crNdv_JiVqqyE9:-KJX96crNdv_JiVqqyEA:1863383193) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.h#L12)):*
> ```objc
> namespace ciface
> {
>     namespace iOS
> ```

We don't put newlines for individual namespaces.

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.h, line 15 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX8xlAiCDTVVnFLYaU:-KJX8xlBZICud1BgwPiN:-240457424) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.h#L15)):*
> ```objc
>     {
> 
>     void Init( std::vector<Core::Device*>& devices );
> ```

Don't put spaces around parameters

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX9IvjYPw-1fULDSTX:-KJX9IvjYPw-1fULDSTY:676036456) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm#L1)):*
> ```XML
> // Copyright 2013 Dolphin Emulator Project
> ```

2016

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm, line 16 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX9FEjK67C1vGZnuxM:-KJX9FEk3SOkJi_PF5-u:-403572188) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm#L16)):*
> ```XML
> {
> 
> void Init( std::vector<Core::Device*>& devices )
> ```

No spaces around parameters

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm, line 175 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX9MoHwfQKM4kmv6d8:-KJX9MoHwfQKM4kmv6d9:-305512373) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm#L175)):*
> ```XML
> {
> 	std::ostringstream ss;
> 	ss << "Button " << (int)_index;
> ```

static_cast

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm, line 186 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX9OPn6kzU_qAn0Za_:-KJX9OPn6kzU_qAn0Zaa:-305512373) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOS.mm#L186)):*
> ```XML
> {
> 	std::ostringstream ss;
> 	ss << "Axis " << (int)_index;
> ```

static_cast

---

*[Source/Core/InputCommon/ControllerInterface/iOS/iOSButtonManager.cpp, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX9SVPOQhv-MEbSJhY:-KJX9SVQ-Ng8snTRF_vT:676036456) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/InputCommon/ControllerInterface/iOS/iOSButtonManager.cpp#L1)):*
> ```cpp
> // Copyright 2013 Dolphin Emulator Project
> ```

2016

---

*[Source/iOS/DolphiniOS/AppDelegate.m, line 20 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXA0doX9Bs4fQNWBUj:-KJXA0doX9Bs4fQNWBUk:513034824) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/AppDelegate.m#L20)):*
> ```objc
> #import "SCLAlertView.h"
> #import "ZAActivityBar.h"
> ```

This file could do with some spacing cleanup.

---

*[Source/iOS/DolphiniOS/EmulatorViewController.mm, line 9 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXAQrCwQbuKwvuO6n6:-KJXAQrDXdWaKTVkZB6V:-646454549) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/EmulatorViewController.mm#L9)):*
> ```XML
> //
> 
> #import "EmulatorViewController.h"
> ```

These paths should be relative like the others

---

*[Source/iOS/DolphiniOS/GameTableView.h, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXAbKCL_A5yHpAa0lR:-KJXAbKDwvOUMzsfQNuX:579120973) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/GameTableView.h#L6)):*
> ```objc
> //
> //  Created by Will Cobb on 5/31/16.
> //  Copyright © 2016 Dolphin. All rights reserved.
> ```

We actually license under GPLv2+, so all rights actually aren't reserved.

---

*[Source/iOS/DolphiniOS/GameTableView.m, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXAplbeBvki4mt_nya:-KJXAplcK73rc0O0LHJ_:-1814689218) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/GameTableView.m#L6)):*
> ```objc
> //
> //  Created by Will Cobb on 11/4/15.
> //  Copyright © 2015 iNDS. All rights reserved.
> ```

Same here (if this is taken from somewhere else, this is problematic)

---

*[Source/iOS/DolphiniOS/GameTableView.m, line 9 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXAvtvKa8HCcnCC8_a:-KJXAvtwDdOibA93L_8q:119988443) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/GameTableView.m#L9)):*
> ```objc
> //
> 
> #import "GameTableView.h"
> ```

Paths should be relative.

---

*[Source/iOS/DolphiniOS/GCAdapteriOS.cpp, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXBCoq2_BssUu1D5a4:-KJXBCoq2_BssUu1D5a5:340977028) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/GCAdapteriOS.cpp#L1)):*
> ```cpp
> // Copyright 2014 Dolphin Emulator Project
> ```

This source file needs to be cleaned up.

---

*[Source/iOS/DolphiniOS/MainiOS.mm, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXBMKxraz5yxrHfo2C:-KJXBMKyusXZaXRBRdsY:160390343) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/MainiOS.mm#L1)):*
> ```XML
> 
> ```

Uh?

---

*[Source/iOS/DolphiniOS/RomTableViewController.h, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXBQETXKQe8mMUXgHq:-KJXBQETXKQe8mMUXgHr:-1974408543) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/RomTableViewController.h#L6)):*
> ```objc
> //
> //  Created by iNDS on 6/9/13.
> //  Copyright (c) 2014 iNDS. All rights reserved.
> ```

Again, we don't reserve all rights.

---

*[Source/iOS/DolphiniOS/RomTableViewController.m, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXBUBWhg8zqKYEreJz:-KJXBUBWhg8zqKYEreK-:-499318459) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/RomTableViewController.m#L6)):*
> ```objc
> //
> //  Created by iNDS on 6/9/13.
> //  Copyright (c) 2014 iNDS. All rights reserved.
> ```

Ditto

---

*[Source/iOS/DolphiniOS/Controller/GCControllerView.m, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXDP-8HAo2lWV-EUgu:-KJXDP-9wuthpoOC0uo9:315800536) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Controller/GCControllerView.m#L6)):*
> ```objc
> //
> //  Created by Will Cobb on 1/18/16.
> //  Copyright © 2016 Will Cobb. All rights reserved.
> ```

Ditto about not reserving all rights

---

*[Source/iOS/DolphiniOS/Controller/GCControllerView.m, line 11 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXDTpLZ5bWRUGMuFcR:-KJXDTpM9UPygcMk0L6b:1936101150) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Controller/GCControllerView.m#L11)):*
> ```objc
> #import "GCControllerView.h"
> #import "WCDirectionalControl.h"
> //#include "InputCommon/GCPadStatus.h"
> ```

Don't leave uncommented code around. Also imports/includes should be relative.

---

*[Source/iOS/DolphiniOS/Controller/WCButtonControl.h, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXDeNQoe7w8Yf_0MUc:-KJXDeNQoe7w8Yf_0MUd:119755402) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Controller/WCButtonControl.h#L6)):*
> ```objc
> //
> //  Created by CC on 7/5/13.
> //  Copyright (c) 2014 CC. All rights reserved.
> ```

Ditto about rights being reserved.

---

*[Source/iOS/DolphiniOS/Controller/WCButtonControl.m, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXDmvXGqFdOy1f5CPx:-KJXDmvXGqFdOy1f5CPy:-499318459) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Controller/WCButtonControl.m#L6)):*
> ```objc
> //
> //  Created by Riley Testut on 7/5/13.
> //  Copyright (c) 2014 CC. All rights reserved.
> ```

Ditto

---

*[Source/iOS/DolphiniOS/Controller/WCDirectionalControl.h, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXDrBzATlpMU0DmTyJ:-KJXDrBzATlpMU0DmTyK:-499318459) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Controller/WCDirectionalControl.h#L6)):*
> ```objc
> //
> //  Created by CC
> //  Copyright (c) 2014 CC. All rights reserved.
> ```

Ditto

---

*[Source/iOS/DolphiniOS/Controller/WCDirectionalControl.m, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXDtEnpjQyYxO8-n4e:-KJXDtEnpjQyYxO8-n4f:-499318459) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Controller/WCDirectionalControl.m#L6)):*
> ```objc
> //
> //  Created by CC
> //  Copyright (c) 2014 CC. All rights reserved.
> ```

Ditto

---

*[Source/iOS/DolphiniOS/Roms/starfield.dol, line 0 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXE7LVKMiscIKmNao1:-KJXE7LVKMiscIKmNao2:7934231) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Roms/starfield.dol#L0)):*
Why is starfield being packaged with source code?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881)*
<!-- Sent from Reviewable.io -->

WilliamLCobb commented:
Review status: 23 of 246 files reviewed at latest revision, 46 unresolved discussions, some commit checks failed.

---

*[Source/Core/Common/Arm64Emitter.cpp, line 15 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWt3v_t0MZqU_SKKG2:-KJX9lm_Vr2zCBts7QgX:1135377602) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/Arm64Emitter.cpp#L15)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> no need to hide array within IOS. It's either needed for all, or non.
</details>
It wouldn't compile without the include on iOS. It wasn't there before

---

*[Source/Core/Common/ArmCPUDetect.cpp, line 66 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWtSxGV1xk1DZGa0CX:-KJXFtC8RNvaczp5wrGw:1202796884) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/ArmCPUDetect.cpp#L66)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> I think we should try to support this. 
</details>
I agree, I'm not sure how though

---

*[Source/Core/Common/Network.h, line 11 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWuijFfdyTdLU6KNge:-KJXG3QTpOEY8RFQYXDN:-837283981) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/Network.h#L11)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> Will an enum class help here?
</details>
Changed back to IOS. I'm use APPLE_IOS for the definition now

---

*[Source/Core/Common/GL/GLInterface/IGL.mm, line 53 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWwepcwVSkF22COGXV:-KJXGDUqBekCT0C6J1h-:1156988051) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/GL/GLInterface/IGL.mm#L53)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> why times 2?
</details>
Size is in points, not pixels. I changed this to make it more cear

---

*[Source/Core/Common/GL/GLInterface/IGL.mm, line 25 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX8_DIsE247xt8BRtp:-KJXGBK6zfvTuYNPx5Rr:610327451) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/Core/Common/GL/GLInterface/IGL.mm#L25)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> Please don't leave commented code all over the place.
</details>
Removed

---

*[Source/Core/Common/Logging/ConsoleListeneriOS.cpp, line 16 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWxRVRttXYeIr4TfM6:-KJXGMmv6YknlMjdHmnt:-197900446) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Common/Logging/ConsoleListeneriOS.cpp#L16)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> I wouldn't call this "ios", more like a generic one. If there isn't one (I haven't checked), please just rename this one to "Generic"
</details>
Removed log file  completely. It ended up not being necessary

---

*[Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp, line 20 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWzyspW5RDyVAmdR9c:-KJXGVAs2EhRgJ4NafpV:-1494858505) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp#L20)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> Did this functions fail to compile, or just fail at execution? I'd like to see fastmem at some point :/
</details>
Failed to compile. MachineContext.h needs to be updated to support iOS

---

*[Source/Core/VideoBackends/OGL/Render.cpp, line 1340 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJX-chYEZCef5qMfbuJ:-KJXHzYFEpNJzo9oN3Cq:-1184515814) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/VideoBackends/OGL/Render.cpp#L1340)):*
<details><summary><i>Previously, degasus (Markus Wick) wrote…</i></summary>
> Maybe within BlitScreen?
</details>
Just let me know where it should go

---

*[Source/iOS/DolphiniOS/EmulatorViewController.mm, line 9 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXAQrCwQbuKwvuO6n6:-KJXIudSjoxwZH9vkWYN:-1729167390) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/EmulatorViewController.mm#L9)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> These paths should be relative like the others
</details>
Relative paths are not generally usually used in Xcode projects. I can definitely change this though if they are preffered

---

*[Source/iOS/DolphiniOS/GameTableView.h, line 6 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXAbKCL_A5yHpAa0lR:-KJXJyMYMaUZpgyU-0tV:1464259854) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/GameTableView.h#L6)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> We actually license under GPLv2+, so all rights actually aren't reserved.
</details>
Xcode automatically adds this. I've removed them

---

*[Source/iOS/DolphiniOS/MainiOS.mm, line 1 \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXBMKxraz5yxrHfo2C:-KJXL2c-BeRFqPHzf0SU:-328310230) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/MainiOS.mm#L1)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> Uh?
</details>
-_- not sure where the source went... I'll add it back

---

*[Source/iOS/DolphiniOS/Roms/starfield.dol, line  \[r3\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJXE7LVKMiscIKmNao1:-KJXLvkKOD7RrWms4AtT:-1054325829) ([raw file](https://github.com/dolphin-emu/dolphin/blob/d43c893571895dcd99d083adc8cf83b550a424fa/Source/iOS/DolphiniOS/Roms/starfield.dol#L)):*
<details><summary><i>Previously, lioncash (Mat M.) wrote…</i></summary>
> Why is starfield being packaged with source code?
</details>
removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881)*
<!-- Sent from Reviewable.io -->

WilliamLCobb commented:
Well I rebased the repo and now it's showing unknown repo in this pr and pushes from my fork aren't being updated. I'll close this and open a new pull request and only use force push from now on, unless someone knows how a way to fix this

OatmealDome commented:
Review status: 23 of 246 files reviewed at latest revision, 46 unresolved discussions, some commit checks failed.

---

*[Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp, line 20 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWzyspW5RDyVAmdR9c:-KJXpR-lwLarZWSfbyEv:656438930) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp#L20)):*
<details><summary><i>Previously, WilliamLCobb (Will Cobb) wrote…</i></summary>
> Failed to compile. MachineContext.h needs to be updated to support iOS
</details>
I haven't tested it, but [here's](https://github.com/OatmealDome/dolphin/commit/b363dc5164a8bd2b75652bfd12e15ea5c11148d3) a MachineContext for iOS if you need it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881)*
<!-- Sent from Reviewable.io -->

WilliamLCobb commented:
Review status: 23 of 246 files reviewed at latest revision, 46 unresolved discussions, some commit checks failed.

---

*[Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp, line 20 \[r2\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881#-KJWzyspW5RDyVAmdR9c:-KJYMTJuoPk7BVVoGdLV:1245673023) ([raw file](https://github.com/dolphin-emu/dolphin/blob/ea7c88c70e85cf0293587ec42741e9b28bca6b48/Source/Core/Core/PowerPC/JitArm64/JitArm64_BackPatch.cpp#L20)):*
<details><summary><i>Previously, OatmealDome wrote…</i></summary>
> I haven't tested it, but [here's](https://github.com/OatmealDome/dolphin/commit/b363dc5164a8bd2b75652bfd12e15ea5c11148d3) a MachineContext for iOS if you need it.
</details>
Awesome! I added it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3881)*
<!-- Sent from Reviewable.io -->

WilliamLCobb commented:
Commits stopped updating to this PR. Moved to https://github.com/dolphin-emu/dolphin/pull/3885
